projects/aca-content/src/lib/components/search/search-input/search-input.component.spec.ts

```diff
@@ -239,4 +239,205 @@ describe('SearchInputComponent', () => {
     });
     fixture.detectChanges();
   });
+
+  describe('searchByOption()', () => {
+    let isLibrariesCheckedSpy: jasmine.Spy;
+    let isFoldersCheckedSpy: jasmine.Spy;
+    let isFilesCheckedSpy: jasmine.Spy;
+    let evaluateLibrariesConstraintSpy: jasmine.Spy;
+    let syncInputValuesSpy: jasmine.Spy;
+    let isSameSearchTermSpy: jasmine.Spy;
+    let queryBuilderUpdateSpy: jasmine.Spy;
+    let queryLibrariesBuilderUpdateSpy: jasmine.Spy;
+    let storeDispatchSpy: jasmine.Spy;
+
+    beforeEach(() => {
+      isLibrariesCheckedSpy = spyOn(component, 'isLibrariesChecked').and.returnValue(false);
+      isFoldersCheckedSpy = spyOn(component, 'isFoldersChecked').and.returnValue(false);
+      isFilesCheckedSpy = spyOn(component, 'isFilesChecked').and.returnValue(false);
+      evaluateLibrariesConstraintSpy = spyOn(component, 'evaluateLibrariesConstraint').and.returnValue(false);
+      syncInputValuesSpy = spyOn(component, 'syncInputValues').and.callThrough();
+      isSameSearchTermSpy = spyOn(component, 'isSameSearchTerm').and.returnValue(false);
+
+      queryBuilderUpdateSpy = spyOn(component['queryBuilder'], 'update');
+      queryLibrariesBuilderUpdateSpy = spyOn(component['queryLibrariesBuilder'], 'update');
```

Comment:
Please use the same approach like suggested here for queryBuilder: 
https://github.com/Alfresco/alfresco-content-app/pull/4251/files#r1858523721